### DISCUSSION TITLE
Changing vimtex options

### DISCUSSION THREAD

**Discussion by patnr** [LazyVim/LazyVim#1928] [Q&A] [answered]
I used to rely on

```vim
" In ~/.vim/ftplugin/tex.vim
let g:vimtex_compiler_latexmk = {'build_dir': '.tex_build' }
```

and am trying to replicate this customisation of vimtex with LazyVim (with the "tex" extras).

I've tried all of the following without success (I want compilation to happen inside a dir called '.tex_build').

```lua
-- In ~/.config/nvim/init.lua
vim.g.vimtex_compiler_latexmk = { build_dir = ".tex_build", }
```

```lua
-- In ~/.config/nvim/ftplugin/tex.lua
vim.g.vimtex_compiler_latexmk = { build_dir = ".tex_build", }
```

```vim
" In ~/.config/nvim/ftplugin/tex.vim
let g:vimtex_compiler_latexmk = {'build_dir': '.tex_build' }
```

```lua
-- In ~/.config/nvim/plugins/misc.lua
  {
    "lervag/vimtex",
    lazy = false, -- lazy-loading will disable inverse search
    init = function()
        vim.g.vimtex_compiler_latexmk = { build_dir = ".tex_build", }
    end,
  },
```

Anybody knows how to make it work?

**Comment by dpetka2001** [comment]
Seems like you can't write dictionary values into Neovim (see [here](https://neovim.io/doc/user/lua.html#lua-vim-variables))
Trying to follow the example listed in the docs, you could do something like
```lua
local my_vimtex = vim.g.vimtex_compiler_latexmk
my_vimtex.build_dir = '.tex_build'
vim.g.vimtex_compiler_latexmk = my_vimtex
```
in the `init` function like you did in your 3rd example. Not sure if it'll work, but it's worth trying out.

**Reply by patnr** [reply]
Thanks, but
- In my example above I was not writing dictionary values -- I created the whole dictionary.
- I tried your suggestion. Since `vim.g.vimtex_compiler_latexmk` does not exist yet, I must create the table.  
  But anyways it does not change the compilation behaviour.

**Reply by dpetka2001** [reply]
I did some digging in vimtex repo and found this [issue](https://github.com/lervag/vimtex/issues/2778). It seems `build_dir` has been renamed to `out_dir`. Maybe that will fix your issue with your initial code?

**Reply by dpetka2001** [reply]
I just did the search but did not actually provide a solution. Would you mind sharing in a post the solution for you and marking your post as the suggested answer, so that users will find the answer more easily?

**Comment by patnr** [comment, marked as answer]
Thanks to @dpetka2001 who found the solution.

Any of the following works

```lua
-- In ~/.config/nvim/init.lua
vim.g.vimtex_compiler_latexmk = { out_dir = ".tex_build", }
```

```vim
" In ~/.config/nvim/ftplugin/tex.vim
let g:vimtex_compiler_latexmk = {'out_dir': '.tex_build' }
```

```lua
-- In ~/.config/nvim/plugins/misc.lua
  {
    "lervag/vimtex",
    init = function()
        vim.g.vimtex_compiler_latexmk = { out_dir = ".tex_build", }
    end,
  },
```
